### PR TITLE
DST-148: add options for llm and embedding

### PR DESCRIPTION
**Pull request by ccheng26**
You'll need a .env file with a Google API key.
You can add options for different llms to try as long as you download them and change the `llm_choice`, we're using Mistral 7b for this project (option 2), you can download it [here](https://mistral.ai/news/announcing-mistral-7b/)

```
pip install -r requirements
python run.py
```

### CODE REVIEW

File: 02-household-queries/run.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Which embedding function would you like to use?
+1. Google Gen AI (default)
+2. all-MiniLM-L6-v2
+      """)
+embedding_choice = input()
+# Set embedding function
+if embedding_choice == 2 or embedding_choice == "all-MiniLM-L6-v2":
+    embeddings = SentenceTransformerEmbeddings(model_name="all-MiniLM-L6-v2")
+else:
+    embeddings = GoogleGenerativeAIEmbeddings(model="models/embedding-001")
```
Comment (yoomlam) [inline 02-household-queries/run.py:23]: Fine for now. Consider integrating these settings/options into `chainlit-household-bot.py` along with an "Ingest" button (in separate PR).

File: 02-household-queries/run.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+dotenv.load_dotenv()
+
+print("""
+Which embedding function would you like to use?
+1. Google Gen AI (default)
+2. all-MiniLM-L6-v2
+      """)
+embedding_choice = input()
+# Set embedding function
+if embedding_choice == 2 or embedding_choice == "all-MiniLM-L6-v2":
```
Comment (yoomlam) [inline 02-household-queries/run.py:20]:
```suggestion
if embedding_choice == "2" or embedding_choice == "all-MiniLM-L6-v2":
```

File: 02-household-queries/run.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+print("""
+Which LLM would you like to use? 
+1. Gemini (default)
+2. Mistral
+      """)
+
+llm_choice = input()
+# Load the models
+if llm_choice == 2 or llm_choice == "Mistral":
```
Comment (yoomlam) [inline 02-household-queries/run.py:33]:
```suggestion
if llm_choice == "2" or llm_choice == "Mistral":
```

File: 02-household-queries/run.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+vectordb=Chroma(embedding_function=embeddings, collection_name="resources", persist_directory="./chroma_db")
+
+print("""
+Initialize DB and retrieve? 
+1. Retrieve only (default)
+2. Init andd retrieve
+3. Init only
+      """)
+run_option = input()
+if run_option == 2:
```
Comment (yoomlam) [inline 02-household-queries/run.py:56]:
```suggestion
if run_option == "2":
```

File: 02-household-queries/run.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    ingest_call(vectordb=vectordb)
+    retrieval_call(llm=llm, vectordb=vectordb)
+elif run_option == 3:
```
Comment (yoomlam) [inline 02-household-queries/run.py:59]:
```suggestion
elif run_option == "3":
```

File: 02-household-queries/run.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+# Set embedding function
+if embedding_choice == 2 or embedding_choice == "all-MiniLM-L6-v2":
+    embeddings = SentenceTransformerEmbeddings(model_name="all-MiniLM-L6-v2")
+else:
+    embeddings = GoogleGenerativeAIEmbeddings(model="models/embedding-001")
+
+print("""
+Which LLM would you like to use? 
+1. Gemini (default)
+2. Mistral
```
Comment (yoomlam) [inline 02-household-queries/run.py:28]:
```suggestion
2. Mistral (via GPT4All)
```
I'll be adding another option for `Mistral (Ollama)`

File: 02-household-queries/run.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                                verbose = True,google_api_key=GOOGLE_API_KEY,
+                                convert_system_message_to_human=True)
+        
+# initialize chroma db
+vectordb=Chroma(embedding_function=embeddings, collection_name="resources", persist_directory="./chroma_db")
+
+print("""
+Initialize DB and retrieve? 
+1. Retrieve only (default)
+2. Init andd retrieve
```
Comment (yoomlam) [inline 02-household-queries/run.py:52]:
```suggestion
2. Ingest and retrieve
```

File: 02-household-queries/run.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+3. Init only
```
Comment (yoomlam) [inline 02-household-queries/run.py:53]:
```suggestion
3. Ingest only
```

**Review by yoomlam — CHANGES_REQUESTED** [review]

**Comment by KevinJBoyer** [comment]
LGTM (with Yoom's changes)!

**Pull request merged**